core/src/main/scala/com.snowplowanalytics.elasticsearch/loader/StdinSourceExecutor.scala

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2014-2016 Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache
+ * License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.
+ *
+ * See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.elasticsearch.loader
+
+// AWS Kinesis Connector libs
+import com.amazonaws.services.kinesis.connectors.KinesisConnectorConfiguration
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// This project
+import sinks._
+import clients._
+import StreamType._
+
+/**
+ *
+ * @param streamType the type of stream, good/bad
+ * @param documentIndex the elasticsearch index name
+ * @param documentType the elasticsearch index type
+ * @param kinesisConfig the KCL configuration
+ * @param goodSink the configured GoodSink
+ * @param badSink the configured BadSink
+ * @param elasticsearchSender function for sending to elasticsearch
+ */
+class StdinSourceExecutor(
+  streamType: StreamType,
+  documentIndex: String,
+  documentType: String,
+  kinesisConfig: KinesisConnectorConfiguration,
+  goodSink: Option[ISink],
+  badSink: ISink,
+  elasticsearchSender: ElasticsearchSender
+) extends Runnable {
+    val transformer = streamType match {
+        case StreamType.Good => new SnowplowElasticsearchTransformer(documentIndex, documentType)
+        case StreamType.Bad => new BadEventTransformer(documentIndex, documentType)
+    }
+
+    def run = for (ln <- scala.io.Source.stdin.getLines) {
+        val emitterInput = transformer.consumeLine(ln)
+        emitterInput._2.bimap(
+          f => badSink.store(BadRow(emitterInput._1, f).toCompactJson, None, false),          
+          s => goodSink match {
+            case Some(gs) => gs.store(s.getSource, None, true)
+            case None => elasticsearchSender.sendToElasticsearch(List(ln -> s.success))
+          }
+        )
+    }
+}
+
```

Comment:
I just think it would be good if everything in the same format. Will turn into the old version if you does not think so